code/modules/power/tesla/energy_ball.dm

```diff
@@ -0,0 +1,94 @@
+/obj/singularity/energy_ball
+	name = "Energy Ball"
+	icon = 'icons/obj/tesla_engine/energy_ball.dmi'
+	icon_state = "energy_ball"
+	pixel_x = -32
+	pixel_y = -32
+	current_size = STAGE_TWO
+	move_self = 1
+	grav_pull = 0
+	contained = 0
+	density = 1
+
+/obj/singularity/energy_ball/process()
+	pixel_x = 0
+	pixel_y = 0 // Lining up the beams properly.
+	tesla_zap(src, 3, 3476520)
+	pixel_x = -32
+	pixel_y = -32
+	var/move_dir = pick(alldirs)
+	var/turf/T = get_step(src,move_dir)
+	if(can_move(T))
+		loc = get_step(src,move_dir)
+	return
+
+/obj/singularity/energy_ball/Bump(atom/A)
+	if(istype(A, /mob/living/carbon))
+		var/mob/living/carbon/C = A
+		C.dust()
+	return
+
+/obj/singularity/energy_ball/Bumped(atom/A)
+	if(istype(A, /mob/living/carbon))
+		var/mob/living/carbon/C = A
+		C.dust()
+	return
+
+proc/get_closest_atom(var/type, var/list, var/source)
```

Comment:
Yeah, i made it specifically for this but moving it to being a helper sounds good